app/Controllers/authController.php

```diff
@@ -125,15 +125,15 @@ public function formLoginAction() {
 
 			FreshRSS_Context::initUser($username);
 			if (FreshRSS_Context::$user_conf == null) {
-				// Initialise the default user to be able to display the error page
-				FreshRSS_Context::initUser(FreshRSS_Context::$system_conf->default_user);
-				Minz_Error::error(403, array(_t('feedback.auth.login.invalid')), false);
+				// Username is unknown
+				Minz_Request::bad(_t('feedback.auth.login.invalid'), ['c' => 'auth', 'a' => 'login']);
```

Comment:
Missing HTTP 403 status

Reply:
I cannot find any way to inform the user with a useful text + send the HTTP403 status at once.
If 403 sent, than only a generic message is shown.

Reply:
We might need something similar than lines 166-169 or expend the Minz functions.
I can have a look, but probably first after we have shipped 1.19.2.

Reply:
Line 168 does not have any effect in this case
https://github.com/FreshRSS/FreshRSS/blob/edge/app/Controllers/authController.php#L168

Reply:
HTTP error 401 would be better than 403. https://developer.mozilla.org/en-US/docs/Web/HTTP/Status#client_error_responses

Reply:
It is not for the same case. 401 is used for HTTP Basic Auth in particular.